bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableTableAdminClientWrapper.java

```diff
@@ -0,0 +1,212 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.grpc;
+
+import com.google.bigtable.admin.v2.DeleteTableRequest;
+import com.google.bigtable.admin.v2.DropRowRangeRequest;
+import com.google.bigtable.admin.v2.GetTableRequest;
+import com.google.bigtable.admin.v2.ListTablesRequest;
+import com.google.bigtable.admin.v2.ListTablesResponse;
+import com.google.cloud.bigtable.admin.v2.models.CreateTableRequest;
+import com.google.cloud.bigtable.admin.v2.models.ModifyColumnFamiliesRequest;
+import com.google.cloud.bigtable.admin.v2.models.Table;
+import com.google.cloud.bigtable.config.BigtableOptions;
+import com.google.cloud.bigtable.core.IBigtableTableAdminClient;
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Empty;
+import java.util.List;
+
+/**
+ * This class implements the {@link IBigtableTableAdminClient} interface which provides access to
+ * google cloud java.
+ */
+public class BigtableTableAdminClientWrapper implements IBigtableTableAdminClient {
+
+  private final BigtableTableAdminClient adminClient;
+  private final BigtableInstanceName instanceName;
+
+  public BigtableTableAdminClientWrapper(BigtableTableAdminClient adminClient,
+      BigtableOptions options){
+    this.adminClient = adminClient;
+    this.instanceName = new BigtableInstanceName(options.getProjectId(),
+        options.getInstanceId());
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Table createTable(CreateTableRequest request) {
+    com.google.bigtable.admin.v2.CreateTableRequest requestProto =
+        request.toProto(instanceName.toAdminInstanceName());
+    adminClient.createTable(requestProto);
+
+    return getTable(requestProto.getTableId());
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public ListenableFuture<Table> createTableAsync(CreateTableRequest request) {
+    com.google.bigtable.admin.v2.CreateTableRequest requestProto =
+        request.toProto(instanceName.toAdminInstanceName());
+
+    return Futures.transform(adminClient.createTableAsync(requestProto),
+        new Function<com.google.bigtable.admin.v2.Table, Table>() {
+          @Override
+          public Table apply(com.google.bigtable.admin.v2.Table table) {
+            return Table.fromProto(table);
+          }
+        }, MoreExecutors.directExecutor());
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Table getTable(String tableId) {
+    GetTableRequest request = GetTableRequest.newBuilder()
```

Comment:
To avoid confusion, I think it would be good to used fully qualified names for all raw proto types while using short names for gcj models (or vice versa)

Reply:
Or at the very least, name the proto variables with a Proto suffix

Reply:
updated variables with Proto suffix.